commands/src/main/java/org/jclouds/karaf/commands/compute/ComputeServiceCreateCommand.java

```diff
@@ -145,8 +145,11 @@ public void run() {
                   String providerValue = EnvHelper.getComputeProvider(provider);
                   String apiValue = EnvHelper.getComputeApi(api);
                   String identityValue = EnvHelper.getComputeIdentity(identity);
-                  String credentialValue = EnvHelper.getComputeCredential(credential);
                   String endpointValue = EnvHelper.getComputeEndpoint(endpoint);
+                  String credentialValue = EnvHelper.getComputeCredential(credential);
+                  if (providerValue != null && credentialValue != null && providerValue.equals("google-compute-engine")) {
+                     credentialValue = EnvHelper.getGoogleCredentialFromJsonFile(credentialValue);
```

Comment:
If I understand this correctly, we're effectively adding fallback logic - "try normal first, if that doesn't work, try the JSON file". Is that the same sequence as in the provide? Is there some property that we could use to determine up front which of the two options is being used?

Also, as a minor change, check `credentialValue != null` first?